### PR TITLE
[depends] - fixes which allow building for android on osx mashines

### PR DESCRIPTION
**Pull request by Memphiz**
Well that one worked at one point shortly after the android port hit mainline and regressed through the time. I picked it up again :)

Most of those commits are self-explaining. Basically we always need to ensure
a) that native stuff is built with native compile flags (see TexturePacker)
b) that target stuff always uses the tools from the toolchain (like AR and RANLIB) - because those are incompatible on osx (thats why it works when compiling on linux hosts because those tools seem to be compatible to the toolchains tools)

@koying and @wsnipex for a look - @koying this removes the ndk_ver sed stuff - i was not able to get this working on osx (sed is such a bitch espacially when called from configure.in - autoconf kept eating the [] chars of the sed).

This also updates the README.android with special section for building on osx (case sensitive filesystem image is needed for example...)

A user in the forum confirmed this branch working on yosemite 10.10 - i confirmed it working on mavericks 10.9.5.

### CODE REVIEW

File: tools/depends/configure.in [hunk 1]
```diff
@@ -91,6 +91,39 @@ passed_ldflags="$LDFLAGS"
 passed_cxxflags="$CXXFLAGS"
 platform_cc=gcc
 platform_cxx=g++
+
+case $build in
+  arm*-*linux-android*)
+    build_os="android"
```
Comment (koying) [inline tools/depends/configure.in]: build_os is... the build os, right?
I very much doubt we'll ever be able to build kodi on android self ;)

Comment (Memphiz) [inline tools/depends/configure.in]: haha - yeah you are right - i basically c&p'ed the case from the $host var below ... you want me to remove it?

Comment (koying) [inline tools/depends/configure.in]: If you have to do other fixes. It might be confusing...

File: tools/depends/configure.in [hunk 2]
```diff
@@ -91,6 +91,39 @@ passed_ldflags="$LDFLAGS"
...
+        build_os="osx"
+      ;;
+
+      arm-apple-darwin*)
+        build_os="ios"
+      ;;
+    esac
+  ;;
+  *)
+    AC_MSG_ERROR(unsupported host ($use_host))
```
Comment (koying) [inline tools/depends/configure.in]: Souldn't it be $build?

Comment (Memphiz) [inline tools/depends/configure.in]: yep - keep going :)

File: m4/xbmc_arch.m4 [hunk 3]
```diff
@@ -1,5 +1,46 @@
...
+  powerpc-*-linux-gnu*|powerpc-*-linux-uclibc*)
+     AC_SUBST(NATIVE_ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX -D_POWERPC")
+     ;;
+  powerpc64-*-linux-gnu*|powerpc64-*-linux-uclibc*)
+     AC_SUBST(NATIVE_ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX -D_POWERPC64")
+     ;;
+  arm*-*-linux-gnu*|arm*-*-linux-uclibc*)
+     AC_SUBST(NATIVE_ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX")
+     ;;
+  *-*linux-android*)
```
Comment (koying) [inline m4/xbmc_arch.m4]: Same as below? Is $build the host (build os) or target?

File: m4/xbmc_arch.m4 [hunk 4]
```diff
@@ -1,5 +1,46 @@
...
+     AC_SUBST(NATIVE_ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX -DTARGET_ANDROID")
+     ;;
+  *)
+     AC_MSG_ERROR(unsupported host ($host))
```
Comment (koying) [inline m4/xbmc_arch.m4]: $build ?

File: m4/xbmc_arch.m4 (hunk 3 above)
Comment (Memphiz) [inline m4/xbmc_arch.m4]: $build is the host/native $host is the target ... yeah bad naming - but i don't think those come from us ;)

**Comment by Memphiz** [comment]
updated and handled comments of @koying

File: m4/xbmc_arch.m4 (hunk 3 above)
Comment (wsnipex) [inline m4/xbmc_arch.m4]: this should have been host = native build machine and target = OS and cpu to build for
right from the beginning. Thats what its called everywhere else. Not your fault though :)

File: docs/README.android [hunk 5]
```diff
@@ -69,7 +95,7 @@ archives to your harddisk. For our example we are extracting in the following
 directories (this matches the example from tools/depends/README aswell):
 
 NDK (referenced as <android-ndk> from now on):
-/opt/android-ndk-r9
+/opt/android-ndk-r9c
```
Comment (wsnipex) [inline docs/README.android]: still r9c? shouldn't this reflect the 10d recommendation above?

File: m4/xbmc_arch.m4 [hunk 6]
```diff
@@ -36,7 +67,7 @@ case $host in
      AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX -DTARGET_ANDROID")
      ;;
   *)
-     AC_MSG_ERROR(unsupported host ($host))
+     AC_MSG_ERROR(unsupported host (target $host))
```
Comment (wsnipex) [inline m4/xbmc_arch.m4]: host = native build machine. could you make it:
AC_MSG_ERROR(unsupported build target: $host)

File: tools/depends/configure.in [hunk 7]
```diff
@@ -91,6 +91,25 @@ passed_ldflags="$LDFLAGS"
...
+    build_os="linux"
+  ;;
+  x86_64*-linux-gnu*|x86_64-*-linux-uclibc*)
+    build_os="linux"
+  ;;
+  *darwin*)
+    build_os="osx"
+  ;;
+  *)
+    AC_MSG_ERROR(unsupported build (native $build))
```
Comment (wsnipex) [inline tools/depends/configure.in]: unsupported native build platform: $build?

**Comment by Memphiz** [comment]
updated and handled comments of @wsnipex

**Comment by wsnipex** [comment]
looks good for me. jenkins build this please

File: tools/depends/configure.in [hunk 8]
```diff
@@ -91,6 +91,25 @@ passed_ldflags="$LDFLAGS"
...
+    build_os="linux"
+  ;;
+  x86_64*-linux-gnu*|x86_64-*-linux-uclibc*)
+    build_os="linux"
+  ;;
+  *darwin*)
+    build_os="osx"
+  ;;
+  *)
+    AC_MSG_ERROR(unsupported native build platform: $build)
```
Comment (wsnipex) [inline tools/depends/configure.in]: you might need to quote this to fix the build errors.

Comment (wsnipex) [inline tools/depends/configure.in]: ;; missing as well

Comment (Memphiz) [inline tools/depends/configure.in]: Nope and nope - error is here:

https://github.com/Memphiz/xbmc/blob/fixnative/m4/xbmc_arch.m4#L30

One ")" to much - i pulled the tools/depends/configure from the workspace and examined the line in question and realised it was in the xbmc_arch.m4 ... :)

**Comment by Memphiz** [comment]
jenkins build this please

**Comment by Memphiz** [comment]
come on bitch i gave you hdd space you give me clean builds jenkins build this please

**Pull request merged**